test/upgrade/background_verification_test.go

```diff
@@ -51,42 +39,14 @@ func TestSkipAtBackgroundVerification(t *testing.T) {
 					// Setup
 					func(c upgrade.Context) {
 						c.Log.Info("Setup 1")
-						go func() {
-							// Log messages before Verify phase.
-							for i := 0; i < bgMessages; i++ {
-								msg := fmt.Sprintf("BeforeVerify %d", i)
-								c.Log.Info(msg)
-								expectedTexts = append(expectedTexts, msg)
-							}
-							close(beforeVerifyCh)
-							<-inVerifyCh
-							// Log messages while Verify phase is in progress.
-							for i := 0; i < bgMessages; i++ {
-								msg := fmt.Sprintf("InVerify %d", i)
-								c.Log.Info(msg)
-								expectedTexts = append(expectedTexts, msg)
-							}
-							close(doneCh)
-						}()
 					},
 					// Verify
 					func(c upgrade.Context) {
-						<-beforeVerifyCh
-						close(inVerifyCh)
-						<-doneCh
 						c.Log.Warn(skipMsg)
 						c.T.Skip(skipMsg)
 						c.Log.Info("Verify 1")
 					},
 				),
-				upgrade.NewBackgroundVerification("ShouldNotBeSkipped",
-					func(c upgrade.Context) {
-						c.Log.Info("Setup 2")
-					},
-					func(c upgrade.Context) {
-						c.Log.Info("Verify 2")
-					},
-				),
```

Comment:
Are we dropping this because it's covered in another test?

Reply:
Two reasons:
1) It's covered by other tests, for example [TestSuiteExecuteEmpty](https://github.com/knative/pkg/blob/main/test/upgrade/functions_test.go#L100) tests successful pass through the handler/BackgroundOperation. And [TestFailAtBackgroundVerification](https://github.com/knative/pkg/blob/main/test/upgrade/background_verification_test.go#L131) uses the NewBackgroundVerification API which is a simple wrapper though (so no extensive testing needed for that).
2) It's also removed to keep just one "continual" test so that it can pass in Github action that runs the unit tests with just 2 logical CPUs. And I don't see an easy way to increase GOMAXPROCS because it uses a shared github action.
From my earlier comments you can see that with 2 CPUs the test suite can only run on continual test in parallel with the main flow.